### PR TITLE
Batchnorm changes: fix default axis and remove workaround for AD crasher

### PR DESCRIPTION
**Pull request by jekbradbury**
The `axis` argument in the batch normalization layers in tf.keras and tf.layers refers to the second of the two axes that should be normalized over (see https://fenghz.github.io/images/2018-4-15/Batch_Norm_Picture.png), and defaults to the last axis (as it typically represents channels), while the first axis is always `0`. We should match that semantics. We can also drop an AD workaround, enabling correct inference behavior.

### CODE REVIEW

**Review by rxwei — APPROVED** [review]

File: Sources/DeepLearning/Layer.swift [hunk 1]
```diff
@@ -170,41 +171,32 @@ public struct BatchNorm<Scalar>: Layer
...
     public func applied(to input: Tensor<Scalar>) -> Tensor<Scalar> {
-        return applyTraining(to: input)
+        if learningPhaseIndicator.training {
+            return applyTraining(to: input)
+        } else {
+            return applyInference(to: input)
+        }
+    }
+
+    public func _vjpApplied(to input: Tensor<Scalar>) ->
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift:183]: Drop `public` and make this `@usableFromInline`

**Comment by rxwei** [comment]
Could you move this PR to the official repo?
